db/models/index.js

```diff
@@ -7,11 +7,7 @@ const User = require("./User");
 Product.belongsTo(Category);
 Category.hasMany(Product);
 
-LineItem.belongsTo(Product);
-Product.hasMany(LineItem);
-
-LineItem.belongsTo(Cart);
-Cart.hasMany(LineItem);
+Product.belongsToMany(Cart, { through: LineItem });
```

Comment:
The LineItem is created when you add a product to the cart, for me it seems product and lineItem is one to one mapping, let's discuss this in class today.

Reply:
belongsToMany will use the through model (LineItems) to create a new record every time a product is placed on a cart. We could use the sequelize methods to do so, or just set the Ids on the line item table. When you run sync with this option turned on, productId and cartId columns are placed on the lineItems table. Take a look in PSQL to see what I'm describing.

Reply:
I understand how line 15 works, however, I do have two questions: 
First, do you need to add Cart.belongsToMany(Product, { through: LineItem });? 
Second, for line 10 to 14 you removed, when you say product hasMany LineItem, does it mean a product can be ordered multiple times by different user, then that product can correspond to different lineItems in the different cart?

Reply:
1. do you need to add Cart.belongsToMany(Product, { through: LineItem });? Not necessarily. It depends on whether you want to use the built in sequelize methods, which is probably not necessary here. When we enter a product into a cart, we should have access to the cartId and the productId. So you could just post to a LineItem route with something like { productId, cartId, quantity, price }.

2. or line 10 to 14 you removed, when you say product hasMany LineItem, does it mean a product can be ordered multiple times by different user, then that product can correspond to different lineItems in the different cart?: In short, yes. I got rid of those lines because they are redundant. Line 15 accomplishes that work. My only goal was to add the necessary columns to the LineItem table. Each time you add a new product, just post to line items with the productId and cartId, and you will have a unique key pair for that product specifically on that cart.

Reply:
@5t4mp5, thanks for the explanation. The code looks good now.